src/scroll-timeline-css.js

```diff
@@ -148,11 +148,28 @@ export function initCSSPolyfill() {
   // because we may lose some of the 'animationstart' events by the time 'load' is completed.
   window.addEventListener('animationstart', (evt) => {
     evt.target.getAnimations().filter(anim => anim.animationName === evt.animationName).forEach(anim => {
-      const result = createScrollTimeline(anim, anim.animationName, evt.target);
-      if (result.timeline && anim.timeline != result.timeline) {
-        const proxyAnimation = new ProxyAnimation(anim, result.timeline, result.animOptions);
+      // Create a cache on the element, to store all Proxy Animations in
+      if (!evt.target.proxiedAnimations) {
+        evt.target.proxiedAnimations = new Map();
```

Comment:
We should not hang new stuff off element as it breaks encapsulation.  We have animation and timeline details to avoid exposing polyfill internals to the outside world.  

Instead we can add a global element map to the polyfill, i.e.:

let elementAnimationsMap = new WeakMap();

and then retrieve the map of proxied animations from there.  Please take a look at how  proxyAnimations.get(...) is used to retrieve animation details without exposing to the "outside world".  We should follow the same pattern here.

Reply:
Addressed in 3fb2b7f03af35061c9b272ce2770c2feb4069839.

Note I could not use a `WeakMap` for the nested cache, as WeakMaps require Objects as their key. As the used `anim` always is a new Object, this didn’t work – it only created more Objects.

Using WeakMap             |  Using Map
:-------------------------:|:-------------------------:
![image](https://user-images.githubusercontent.com/213073/195147749-9b65634f-216e-4f4e-b2c9-829b7f091281.png) |  ![image](https://user-images.githubusercontent.com/213073/195147834-a30f43d1-e2d7-4cbd-ae6c-997cd89988e1.png)